src/main/java/org/ice4j/socket/MultiplexingDatagramSocket.java

```diff
@@ -274,6 +384,26 @@ public MultiplexedDatagramSocket getSocket(
     }
 
     /**
+     * Adds a specific {@link MultiplexedDatagramSocket} to this instance. The
+     * given socket must be already associated with this instance.
+     * @param multiplexed the socket to add.
+     * @throws IllegalArgumentException if the given socket is not associated
+     * with this instance, or its filter is already used for another multiplexed
+     * socket.
+     */
+    public void addSocket(MultiplexedDatagramSocket multiplexed)
```

Comment:
i'm confused by the need for this...if it must already be associated with this, why do we allow it to become disassociated?